ui/src/app/applications/components/application-resource-tree/application-resource-tree.tsx

```diff
@@ -116,7 +116,22 @@ function filterGraph(app: models.Application, filteredIndicatorParent: string, g
 }
 
 function compareNodes(first: ResourceTreeNode, second: ResourceTreeNode) {
-    return `${(first.orphaned && '1') || '0'}/${nodeKey(first)}`.localeCompare(`${(second.orphaned && '1') || '0'}/${nodeKey(second)}`);
+    function orphanedToInt(orphaned?: boolean) {
+        return (orphaned && 1) || 0;
+    }
+    function compareRevision(a: string, b: string) {
+        const numberA = Number(a);
+        const numberB = Number(b);
+        if (isNaN(numberA) || isNaN(numberB)) {
+            return a.localeCompare(b);
+        }
+        return Math.sign(numberA - numberB);
+    }
+    return (
+        orphanedToInt(first.orphaned) - orphanedToInt(second.orphaned) ||
+        compareRevision(first.resourceVersion, second.resourceVersion) ||
```

Comment:
The revision is stored in `info` field, not in `resourceVersion`. Also it is available only for some resources so we need to check if `info` has it.

https://github.com/argoproj/argo-cd/blob/9f47a116219700a46750707fd4d0e5f835404444/ui/src/app/shared/models.ts#L324

Here is sample resource:

```json
{
    "group": "apps",
    "version": "v1",
    "kind": "Deployment",
    "namespace": "argocd",
    "name": "argocd-redis-ha-haproxy",
    "uid": "6492f1e5-3301-4d02-adca-7c6a63109840",
    "info": [
        {
            "name": "Revision",
            "value": "Rev:1"
        }
    ],
    "resourceVersion": "319468248",
    "health": {
        "status": "Healthy"
    },
    "createdAt": "2020-06-10T00:54:59Z"
}
```